src/lint/index.ts

```diff
@@ -0,0 +1,27 @@
+import { defaultRules } from "./rules";
+import { RuleCheckResults } from "types/rules";
+import parse from "@commitlint/parse";
+
+export type LintResults = LintResult[];
+export interface LintResult {
```

Comment:
I think a better design would be:
- `lintCommitMessage` takes in a commit message
-  it parse the commit and pass it to `rule.check`
- `rule.check` returns a `Violation` if it fails, or null if acceptable.
- it collects a list of `Violation` from result of `rule.check`
- it return an aggregated score and the violations.

Therefore, the commit message lint logic resides within the `lint` module, and caller does not need to aware of the internal linting logic, such as score aggregation.